mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
>silent stuff is always scary

Yes, that's why I'm warning here!

>But you can sort the unique event keys and then assign them trigger codes? 

What if one subject has an S1 and the other doesn't? Then the mapping will differ for every single trigger, and it will be very complicated to work across subjects.

>or maybe look at hashlib if it can generate integer hashes ... that could be another way to go

The problem with hashing is they're unpredictable. This way, if a stimulus is called "S196" or "D196", it will be 196 - this is, I think, the least surprising solution. You don't have to look up what hash 8sj@us is, you just know "S196" == 196.